src/benchmark_ocl.cpp

```diff
@@ -144,125 +119,106 @@ int main(void)
-		ocl_error_handler(err, "clEnqueueWriteBuffer(hamiltonian)");
+		err = ocl_helper.queue().enqueueWriteBuffer(hamiltonian_ocl, CL_TRUE, 0, size_hamiltonian_byte, hamiltonian);
+		noma::ocl::error_handler(err, "enqueueWriteBuffer(hamiltonian)");
 	}; // write_hamiltonian
 
 	// lambda to write data to the device
 	auto write_sigma = [&]()
 	{
 		// write data to device
-		err = clEnqueueWriteBuffer(CLU_DEFAULT_Q, sigma_in_ocl, CL_TRUE, 0, size_sigma_byte, sigma_in, 0, nullptr, nullptr);
-		ocl_error_handler(err, "clEnqueueWriteBuffer(sigma_in_ocl)");
-		err = clEnqueueWriteBuffer(CLU_DEFAULT_Q, sigma_out_ocl, CL_TRUE, 0, size_sigma_byte, sigma_out, 0, nullptr, nullptr);
-		ocl_error_handler(err, "clEnqueueWriteBuffer(sigma_out_ocl)");
+		err = ocl_helper.queue().enqueueWriteBuffer(sigma_in_ocl, CL_TRUE, 0, size_sigma_byte, sigma_in);
+		noma::ocl::error_handler(err, "enqueueWriteBuffer(sigma_in_ocl)");
+		err = ocl_helper.queue().enqueueWriteBuffer(sigma_out_ocl, CL_TRUE, 0, size_sigma_byte, sigma_out);
+		noma::ocl::error_handler(err, "enqueueWriteBuffer(sigma_out_ocl)");
 	}; // write_sigma
 
 	// lambda to get the result from the device and compare it with the reference
 	auto read_and_compare_sigma = [&]()
 	{
 		// read data from device	
-		err = clEnqueueReadBuffer(CLU_DEFAULT_Q, sigma_out_ocl, CL_TRUE, 0, size_sigma_byte, sigma_out, 0, nullptr, nullptr);
-		ocl_error_handler(err, "clEnqueueReadBuffer(sigma_out_ocl)");
+		err = ocl_helper.queue().enqueueReadBuffer(sigma_out_ocl, CL_TRUE, 0, size_sigma_byte, sigma_out);
+		noma::ocl::error_handler(err, "clEnqueueReadBuffer(sigma_out_ocl)");
 		// compute deviation from reference	(small deviations are expected)
 		deviation = compare_matrices(sigma_out, sigma_reference_transformed, dim, num);
 		std::cerr << "Deviation:\t" << deviation << std::endl;
 	}; // read_and_compare_sigma
 
 	// Lambda to: transform memory, benchmark, compare results
 	// NOTE:
-	// typedef struct
+	// struct noma::ocl::nd_range
 	// {
-	//     cl_uint   dim; 
-	//     size_t    global[3];
-	//     size_t    local[3];
-	//     size_t    offset[3];
-	// } clu_nd_range;
+	//     cl::NDRange offset;
+	//     cl::NDRange global;
+	//     cl::NDRange local;
+	// };
 	auto benchmark = [&](const std::string& file_name, const std::string& kernel_name,
-	                     const std::string& compile_options, size_t vec_length, clu_nd_range range,
+	                     const std::string& compile_options, size_t vec_length, noma::ocl::nd_range& range,
 	                     decltype(&transform_matrices_aos_to_aosoa) transformation_sigma,
 	                     bool scale_hamiltonian,
-	                     decltype(&transform_matrix_aos_to_soa) transformation_hamiltonian)
+	                     decltype(&transform_matrix_aos_to_soa) transformation_hamiltonian,
+	                     noma::ocl::helper& ocl_helper)
```

Comment:
we can remove `ocl_helper` from the argument list, as we capture all variables of `benchmark`'s surrounding code by reference `[&]` (Google C++ Lambda or check cppreference.com for further input). We only need arguments that differ during calls.

Reply:
Done with fa04e8c75f867e6b7b543462a86dc012327d7fd3.